src/api.ts

```
@@ -1,10 +1,11 @@
 import { MovieData } from './types/MovieData';
-import { ResponseError } from './types/ReponseError';
+import { ResponseError } from './types/ResponseError';
 
-const API_URL = 'https://www.omdbapi.com/?apikey=your-key';
+const API_KEY = import.meta.env.VITE_OMDB_API_KEY as string;
```

Good: reading the API key from the Vite env variable is correct. Consider adding an early runtime check to fail fast with a clear message when `VITE_OMDB_API_KEY` is not provided (helps debugging in local/dev/CI). For example, throw an Error or `console.error` when `API_KEY` is falsy so consumers of `getMovie` get a clear signal instead of a generic fetch/network error.